src/switch_core_media.c

```diff
@@ -1871,9 +1891,12 @@ static void switch_core_session_parse_crypto_prefs(switch_core_session_t *sessio
 		var = "rtp_secure_media_outbound";
 	}
 
-	if (!(val = switch_channel_get_variable(session->channel, var))) {
+	if (!(val = switch_channel_get_variable(session->channel, var)) || !zstr(smh->mparams->rtp_secure_media)) {
```

Comment:
The validation for rtp_secure_media for smh should be done in the if statement below.